examples/direct_fidelity_estimation.py

```diff
@@ -49,9 +50,32 @@ def compute_characteristic_function(circuit: cirq.Circuit,
     return trace, prob
 
 
+async def estimate_characteristic_function(
+        circuit: cirq.Circuit, P_i: Tuple[cirq.Gate, ...],
+        qubits: List[cirq.Qid], simulator: cirq.DensityMatrixSimulator,
+        samples_per_term: int):
+    n_qubits = len(P_i)
+
+    pauli_string = cirq.PauliString(dict(zip(qubits, P_i)))
+    qubit_map = dict(zip(qubits, range(n_qubits)))
+
+    p = cirq.PauliSumCollector(circuit=circuit,
+                               observable=pauli_string,
+                               samples_per_term=samples_per_term)
+
+    completion = p.collect_async(sampler=simulator)
+    assert await completion is None
```

Comment:
Is there a reason you do this? Why not just `await p.collect_async(sampler=simulator)`

Reply:
I had seen it done this way in the unit test. Changed.